kjerneinfo/kjerneinfo-consumer/src/main/java/no/nav/kjerneinfo/consumer/organisasjon/OrganisasjonRestClientImpl.kt

```diff
@@ -0,0 +1,88 @@
+package no.nav.kjerneinfo.consumer.organisasjon
+
+import com.google.gson.GsonBuilder
+import no.nav.common.log.MDCConstants
+import no.nav.common.rest.client.RestClient
+import no.nav.common.sts.SystemUserTokenProvider
+import no.nav.common.utils.EnvironmentUtils
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.utils.RestConstants
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.utils.TjenestekallLogger
+import okhttp3.Request
+import okhttp3.Response
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import org.springframework.beans.factory.annotation.Autowired
+import java.util.*
+
+interface OrganisasjonV1RestClient {
+    fun hentKjernInfoFraRestClient(orgnummer: String): OrganisasjonResponse
+}
+
+open class OrganisasjonRestClientImpl @Autowired constructor(
+        var stsService: SystemUserTokenProvider
+) : OrganisasjonV1RestClient {
+    val ORG_BASEURL = EnvironmentUtils.getRequiredProperty("EREG_ENDPOINTURL")
+    val url = ORG_BASEURL + "api/v1/organisasjon/"
+    private val log = LoggerFactory.getLogger(OrganisasjonRestClientImpl::class.java)
+    private val gson = GsonBuilder().setDateFormat("yyyy-MM-dd").create()
+
+
+    override fun hentKjernInfoFraRestClient(orgnummer: String): OrganisasjonResponse {
+        val uuid = UUID.randomUUID()
+        try {
+
+            val consumerOidcToken: String = stsService.systemUserToken
+            TjenestekallLogger.info("Oppgaver-request: $uuid", mapOf(
+                    "orgnummer" to orgnummer,
+                    "callId" to MDC.get(MDCConstants.MDC_CALL_ID)
+            ))
+            val response: Response = RestClient.baseClient()
+                    .newCall(
+                            Request.Builder()
+                                    .url("$url$orgnummer/noekkelinfo")
+                                    .header(RestConstants.NAV_CALL_ID_HEADER, MDC.get(MDCConstants.MDC_CALL_ID))
+                                    .header("X-Correlation-ID", MDC.get(MDCConstants.MDC_CALL_ID))
+                                    .header(RestConstants.AUTHORIZATION, RestConstants.AUTH_METHOD_BEARER + RestConstants.AUTH_SEPERATOR + consumerOidcToken)
+                                    .header(RestConstants.NAV_CONSUMER_TOKEN_HEADER, RestConstants.AUTH_METHOD_BEARER + RestConstants.AUTH_SEPERATOR + consumerOidcToken)
+                                    .header("accept", "application/json")
+                                    .build()
+                    )
+                    .execute()
+            val body = response.body()?.string()
+            validateResponse(response, uuid, body, orgnummer)
+            return gson.fromJson(body, OrganisasjonResponse::class.java)
```

Comment:
Kan vi bytte til jackson her? Vet vi har gson i bruk andre steder, men ønsker at vi skal komme oss over til ett bibliotek for serialisering/deserialisering og jackson er mest i brukt rundt omkring.

Reply:
Tror vi også må sjekke hva som skjer her om man søker på ett orgnummer som ikke eksisterer.